sklearn/linear_model/tests/test_sgd.py

```diff
@@ -1159,3 +1161,106 @@ def test_large_regularization():
         with np.errstate(all='raise'):
             model.fit(iris.data, iris.target)
         assert_array_almost_equal(model.coef_, np.zeros_like(model.coef_))
+
+
+def _test_gradient_common(loss_function, cases):
+    # Test gradient of different loss functions
+    # cases is a list of (p, y, expected)
+    for p, y, expected in cases:
+        assert_almost_equal(loss_function.dloss(p, y), expected)
+
+
+def test_gradient_hinge():
+    # Test Hinge (hinge / perceptron)
+    loss = sgd_fast.Hinge(0.0)
```

Comment:
Could you also please add tests for `sgd_fast.Hinge(1.0)` which is the most common setting for SVMs?